app/services/forest_liana/schema_utils.rb

```diff
@@ -50,7 +50,7 @@ def self.find_model_from_abstract_class(abstract_class, table_name)
         if subclass.abstract_class?
           return self.find_model_from_table_name(subclass, table_name)
         else
-          subclass.table_name == table_name
+          ForestLiana.name_for(subclass) == table_name
```

Comment:
There is potential issues here since `ForestLiana.name_for` won't necessarily send a table_name anymore.